src/api.c

```diff
@@ -2919,8 +2924,20 @@ int cgroup_create_cgroup(struct cgroup *cgroup, int ignore_ownership)
 	 */
 	for (i = 0; i < cgroup->index; i++) {
 		error = _cgroup_create_cgroup(cgroup, cgroup->controller[i], ignore_ownership);
-		if (error)
+		if (error) {
+			int del_error;
+
+			/*
+			 * This will remove any cgroup directories that were made, but it won't
+			 * undo changes that have been written to the parent cgroup's
+			 * subtree_control file.  To safely undo changes there, we would need to
+			 * save the subtree_control file's previous value and restore it.
+			 */
+			del_error = cgroup_delete_cgroup(cgroup, 1);
+			if (del_error)
+				cgroup_err("Failed to delete %s: %s\n", cgroup->name, del_error);
```

Comment:
@drakenclimber The patch looks good to me. A minor nitpick, in the above error message the format string of `del_error` is of string, was the intention to print `cgroup_strerror(del_error))`

Reply:
Good find.  Thanks